src/tools/kotlin-type-generator.ts

```diff
@@ -36,7 +40,11 @@ export async function generateConnectionSpecType(connection: HandleConnectionSpe
     if (!type.isEntity) return null;
     if (connection.type.hasVariable) return null;
     const node = nodes.find(n => n.schema.equals(type.getEntitySchema()));
-    return ktUtils.applyFun('arcs.core.data.EntityType', [`${node.fullName(connection)}.SCHEMA`]);
+    let schemaReference = `${node.fullName(connection)}.SCHEMA`;
+    if (node.particleSpec.manifestNamespace && node.particleSpec.manifestNamespace !== context.namespace) {
+      schemaReference = `${node.particleSpec.manifestNamespace}.${schemaReference}`;
```

Comment:
In my mind, I though that implementing this would involve adding some sort of import mechanism to the top of the file. I like this though -- adding fully qualified paths reduces code complexity and reduces changes of conflict.

Reply:
That's what I thought as well - I started thinking about imports, but imports make all the codegen decisions not local, which increases complexity of code. Also if you import, then you have opportunity for clashes, which can be a headache as well. This is definitely better I think.